store/tikv/gcworker/gc_worker.go

```diff
@@ -467,11 +477,17 @@ func resolveLocks(ctx goctx.Context, store tikv.Storage, safePoint uint64, ident
 			}
 			continue
 		}
-		regions++
+
 		totalResolvedLocks += len(locks)
-		key = loc.EndKey
-		if len(key) == 0 {
-			break
+		if len(locks) < gcScanLockLimit {
+			regions++
+			key = loc.EndKey
+			if len(key) == 0 {
+				break
+			}
+		} else {
+			// if len(locks) is '0', we should get into the branch above, not here.
```

Comment:
Redundant comments, the code has talked.

Reply:
ok